### PR TITLE
Handle timeout units in doc/v1, and catch parse errors in handler

### PR DESCRIPTION
**Pull request by jonmv**
@hmusum or @baldersheim please review.

### CODE REVIEW

File: vespaclient-container-plugin/src/main/java/com/yahoo/document/restapi/resource/DocumentV1ApiHandler.java [hunk 1]
```diff
@@ -113,7 +114,11 @@
 
     private static final Logger log = Logger.getLogger(DocumentV1ApiHandler.class.getName());
     private static final Parser<Integer> integerParser = Integer::parseInt;
-    private static final Parser<Double> doubleParser = Double::parseDouble;
+    private static final Parser<Double> timeoutParser = value -> {
+        int suffixLength = value.endsWith("ms") ? 2 : value.endsWith("s") ? 1 : 0;
```
Comment (baldersheim) [inline vespaclient-container-plugin/src/main/java/com/yahoo/document/restapi/resource/DocumentV1ApiHandler.java:118]: What about us and ns ?
Don't we have som utility method for this ?

Comment (jonmv) [reply vespaclient-container-plugin/src/main/java/com/yahoo/document/restapi/resource/DocumentV1ApiHandler.java:118]: It doesn't need to be an integer, do I don't see the need for other units. The search handler recognises `ms`, and treats _everything else_ as `s`. I think this is strictly better.

Comment (jonmv) [reply vespaclient-container-plugin/src/main/java/com/yahoo/document/restapi/resource/DocumentV1ApiHandler.java:118]: Actually, it recognise `ks` and `µs` as well, which is just obscene. I found the code, which is split out somewhere, but it's in the `search` module.

Comment (jonmv) [reply vespaclient-container-plugin/src/main/java/com/yahoo/document/restapi/resource/DocumentV1ApiHandler.java]: May as well use that code, then.

Comment (jonmv) [reply vespaclient-container-plugin/src/main/java/com/yahoo/document/restapi/resource/DocumentV1ApiHandler.java]: (It was already a dependency of this module.)

**Comment by jonmv** [comment]
Merging to fix that test.

**Pull request merged**